src/EditorFeatures/Next/Structure/BlockContextProvider.cs

```diff
@@ -135,16 +136,24 @@ private ITextBuffer CreateProjectionBufferForBlockHeaders()
 
                 for (var i = 0; i < blockTags.Count; i++)
                 {
+                    var blockTag = blockTags[i];
+                    var fullStatementSpan = blockTag.StatementSpan;
+                    
+                    if (blockTag.Parent != null &&
+                        textSnapshot.AreOnSameLine(fullStatementSpan.Start, blockTag.Parent.StatementSpan.Start))
```

Comment:
This code could be a little more efficient if you do this check later (after new line 156), keep track of the last line number added, and then skip if the new statementLine.LineNumber <= lastLineNumberAdded.